security/pkg/pki/ca/ca.go

```diff
@@ -330,7 +330,15 @@ func updateCertInConfigmap(namespace string, client corev1.CoreV1Interface, cert
 // returns the certificate chain and the private key.
 func (ca *IstioCA) GenKeyCert(hostnames []string, certTTL time.Duration) ([]byte, []byte, error) {
 	opts := util.CertOptions{
-		RSAKeySize: 2048,
+		RSAKeySize: caKeySize,
```

Comment:
Can we make the RSAKeySize a parameter to the function?

Reply:
GenKeyCert is not for CA key/cert generate only, but you are applying the caKeySize here.

Reply:
I pass it in now as a parameter. Would it make sense to put a reason why 2048 was chosen? I saw magic numbers and wanted to tie it a single var/const.

Reply:
BTW, we should do a check that the key size is never smaller than 2048. Otherwise it's insecure. https://en.wikipedia.org/wiki/Key_size

Reply:
Wrote up issue https://github.com/istio/istio/issues/23426 to track that. That's could be done in a different PR.